lib/origin_simulator.ex

```diff
@@ -39,31 +41,54 @@ defmodule OriginSimulator do
     Process.sleep(10)
 
     recipe = Recipe.parse(Plug.Conn.read_body(conn))
-    Simulation.add_recipe(:simulation, recipe)
+    response = Simulation.add_recipe(:simulation, recipe)
+
+    {status, body, content_type} =
+      case response do
+        :ok -> {201, Poison.encode!(Simulation.recipe(:simulation)), "application/json"}
+        :error -> {406, "Not Acceptable", "text/html"}
+      end
 
     conn
-    |> put_resp_content_type("application/json")
-    |> send_resp(201, Poison.encode!(Simulation.recipe(:simulation)))
+    |> put_resp_content_type(content_type)
+    |> send_resp(status, body)
   end
 
   get "/*glob" do
-    serve_payload(conn)
+    serve_payload_for_route(conn, Simulation.route(:simulation), conn.request_path)
   end
 
   post "/*glob" do
-    serve_payload(conn)
+    serve_payload_for_route(conn, Simulation.route(:simulation), conn.request_path)
   end
 
   match _ do
     send_resp(conn, 404, "not found")
   end
 
-  defp serve_payload(conn) do
+  defp serve_payload_for_route(conn, @default_route, _), do: serve_payload(conn)
+
+  defp serve_payload_for_route(conn, route, path) when route == path,
+    do: serve_payload(conn, route)
+
+  defp serve_payload_for_route(conn, route, path) do
+    cond do
+      # wildcard regex matching
```

Comment:
We probably don't need the comment here, it's clear enough I think.

Reply:
These functions no longer exist (next PR). Since the root wildcard `/*` becomes default route with a standard response, every request that matches `get "/*glob"` will now be served.

Reply:
Yep, that's cool. It's more of a general thing regarding comments as we try to avoid them as much as possible.